pkg/metrics/rate.go

```diff
@@ -35,11 +35,13 @@ func (r *Rate) flush(timestamp float64) ([]*Serie, error) {
 	}
 
 	value, ts := (r.sample-r.previousSample)/(r.timestamp-r.previousTimestamp), r.timestamp
+
+	origSample, origPreviousSample := r.sample, r.previousSample
```

Comment:
Was hoping to avoid having temp variables, but other options would involve changing the logic structure around which probably isn't a good idea ..